host_applications/linux/apps/raspicam/RaspiVid.c

```diff
@@ -2295,15 +2295,46 @@ static int wait_for_next_change(RASPIVID_STATE *state)
       char ch;
 
       if (state->verbose)
-         fprintf(stderr, "Press Enter to %s, X then ENTER to exit\n", state->bCapturing ? "pause" : "capture");
+         fprintf(stderr, "Press Enter to %s, X then ENTER to exit, [i,o,r] then ENTER to change zoom\n", state->bCapturing ? "pause" : "capture");
 
       ch = getchar();
       if (ch == 'x' || ch == 'X')
          return 0;
-      else
-         return keep_running;
+      else if (ch == 'i' || ch == 'I')
+      {
+         if (state->verbose)
+            fprintf(stderr, "Starting zoom in\n");
+
+         raspicamcontrol_zoom_in_zoom_out(state->camera_component, ZOOM_IN, &(state->camera_parameters).roi);
```

Comment:
&state->camera_parameters.roi is sufficient, it doesn't need brackets. Likewise below.